src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java

```diff
@@ -349,69 +349,107 @@ public static PackageMetadataConfig createFromString(String yamlContents) {
     return packageDependencies;
   }
 
-  private static Map<TargetLanguage, VersionBound> createVersionMap(
-      Map<String, Map<String, String>> inputMap) {
-    Map<TargetLanguage, Map<String, String>> intermediate = buildMapWithDefault(inputMap);
-    // Convert parsed YAML map into VersionBound object
-    return Maps.transformValues(
-        intermediate,
-        new Function<Map<String, String>, VersionBound>() {
-          @Override
-          @Nullable
-          public VersionBound apply(@Nullable Map<String, String> versionMap) {
-            if (versionMap == null) {
-              return null;
-            }
-            return VersionBound.create(versionMap.get("lower"), versionMap.get("upper"));
-          }
-        });
+  public static PackageMetadataConfig createFromPackaging(
+      ApiDefaultsConfig apiDefaultsConfig,
+      DependenciesConfig dependenciesConfig,
+      PackagingConfig packagingConfig) {
+    // dependencies config
+    Builder builder =
+        newBuilder()
+            .gaxVersionBound(dependenciesConfig.gaxVersionBound())
+            .gaxGrpcVersionBound(dependenciesConfig.gaxGrpcVersionBound())
+            .gaxHttpVersionBound(dependenciesConfig.gaxHttpVersionBound())
+            .grpcVersionBound(dependenciesConfig.grpcVersionBound())
+            .protoVersionBound(dependenciesConfig.protoVersionBound())
+            .authVersionBound(dependenciesConfig.authVersionBound())
+            .apiCommonVersionBound(dependenciesConfig.apiCommonVersionBound());
+
+    // api_defaults config
+    builder
+        .generatedNonGAPackageVersionBound(apiDefaultsConfig.generatedNonGAPackageVersionBound())
+        .author(apiDefaultsConfig.author())
```

Comment:
A question about the general approach here. We have had to introduce a yaml file and an autovalue file with non-trivial logic that can parse it, and contains a hardcoded reference to that particular yaml file, just to get these values. Can we just make them constants? We could have an ApiDefaultsConfig static class to hold them. The same applies to DependenciesConfig.

Reply:
I agree with the sentiment, but I was aiming for a relatively straightforward move for this step. The status quo is already to parse these values; I'm now having it parse the values from a different location, so this change equates to moving the non-trivial logic from one place to another. 

I actually want to go even further that what you're talking about - for several of the defaults, I don't think we should even have them at all; I think we should require them to be passed in (e.g. "author" doesn't make sense to default IMO). For others, we probably don't need to hold them in a constants class - we can probably move them even closer to where they're used.

Reply:
Fair enough, and I don't want to block on this. If this is actually simpler that putting those values into constants somewhere, then we can make that change later and more thoroughly, as you suggest.